Use test scope in passport tests

hbenl opened the pull request.

bvaughn commented on packages/e2e-tests/testFixtureCloneRecording.ts:
```diff
@@ -72,6 +76,9 @@ const testWithCloneRecording = base.extend<TestIsolatedRecordingFixture>({
         await deleteTestRecording(newRecordingId);
       }
+      if (testUser && testScope) {
+        await resetTestUser(testUser, testScope);
```
Should we log a warning if a `testUser` is supplied without a `testScope`?

Or maybe better yet, can we update the type so it that it requires neither or both?

```ts
type TestIsolatedRecordingFixture = {
  exampleKey: TestRecordingKey;
  pageWithMeta: {
    page: Page;
    recordingId: string;
  };
};

type ScopedTestIsolatedRecordingFixture = TestIsolatedRecordingFixture & {
  testScope: string;
  testUser: string;
};

const testWithCloneRecording = base.extend<
  TestIsolatedRecordingFixture | ScopedTestIsolatedRecordingFixture
>({
```

That way we won't accidentally add any un-scoped authenticated tests?

bvaughn commented on packages/e2e-tests/authenticated/passport-01.test.ts:
```diff
@@ -1,21 +1,27 @@
-  await startTest(page, recordingId, E2E_USER_1_API_KEY);
+  const queryParams = new URLSearchParams();
+  queryParams.set("testScope", testScope!);
+  await startTest(page, recordingId, E2E_USER_1_API_KEY, queryParams);
```
Can we update the `startTest` helper methods to _require_ an API key always include a scope argument as well? e.g.

```ts
async function startTest(
  page: Page,
  recordingId: string
): Promise<void>;

async function startTest(
  page: Page,
  recordingId: string,
  apiKey: string,
  testScope: string
): Promise<void>;

async function startTest(
  page: Page,
  recordingId: string,
  apiKey?: string,
  testScope?: string
) {
  // TS will error if this is called with 3 args
}
```

hbenl commented:
- added the test user emails to `helpers/authentication.ts`
- moved generating the `testScope` to the fixture
- added the ability to reset multiple test users after a test
- moved adding the `testScope` query param to `startTest()`
- changed the `startTest()` signature so that callers can only pass both `apiKey` and `testScope` or none of the two

bvaughn commented on packages/e2e-tests/helpers/utils.ts:
```diff
@@ -197,12 +197,18 @@ export async function locatorTextToNumber(locator: Locator): Promise<number | nu
 }
 
-export async function resetTestUser(email: string) {
+export async function resetTestUser(email: string, testScope?: string) {
```
We should probably require `testScope` since we know that calling this method _without_ a scope can break other running tests.
